string activation
---
string activation_return